lib/SILOptimizer/Mandatory/DIMemoryUseCollectorOwnership.cpp

```diff
@@ -1684,15 +1686,9 @@ void DelegatingInitElementUseCollector::collectClassInitSelfUses() {
   }
 }
 
-void DelegatingInitElementUseCollector::collectValueTypeInitSelfUses() {
-  // When we're analyzing a delegating constructor, we aren't field sensitive at
-  // all.  Just treat all members of self as uses of the single
-  // non-field-sensitive value.
-  assert(TheMemory.NumElements == 1 && "delegating inits only have 1 bit");
-
-  auto *MUI = cast<MarkUninitializedInst>(TheMemory.MemoryInst);
-
-  for (auto UI : MUI->getUses()) {
+void DelegatingInitElementUseCollector::collectValueTypeInitSelfUses(
+    SILInstruction *I) {
+  for (auto UI : I->getResults()[0]->getUses()) {
```

Comment:
Please do not do this! If you truly want a SingleValueInstruction, dyn_cast to it. That being said, here you should be handling multiple results.